src/Storage/Device/S3.php

```diff
@@ -165,14 +169,124 @@ public function getPath(string $filename): string
      *
      * @param string $source
      * @param string $path
+     * @param int chunk
+     * @param int chunks
+     * @param array $metadata
      *
      * @throws \Exception
      *
+     * @return int
+     */
+    public function upload($source, $path, $chunk = 1, $chunks = 1, &$metadata = []): int
+    {
+        if($chunk == 1 && $chunks == 1) {
```

Comment:
Would strict matching `===` be appropriate here?

Reply:
+1